x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -109,14 +111,30 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 
 	switch o.getProvider() {
 	case oAuth2ProviderAzure, oAuth2ProviderDefault:
-		creds := clientcredentials.Config{
-			ClientID:       o.ClientID,
-			ClientSecret:   o.ClientSecret,
-			TokenURL:       o.getTokenURL(),
-			Scopes:         o.Scopes,
-			EndpointParams: o.getEndpointParams(),
+		if o.User != "" || o.Password != "" {
+			conf := &oauth2.Config{
+				ClientID:     o.ClientID,
+				ClientSecret: o.ClientSecret,
+				Endpoint: oauth2.Endpoint{
+					TokenURL:  o.TokenURL,
+					AuthStyle: 1,
+				},
+			}
+			token, err := conf.PasswordCredentialsToken(ctx, o.User, o.Password)
```

Comment:
Any ideas for covering this logic with unit tests?

Reply:
as an idea, if we want to test this bit in isolation, could get extracted to its own function and validate the resulting token